### PR TITLE
Refacto/use flag package

### PR DESCRIPTION
**Pull request by half-shell**
# TODO
- [x] Handle duplicate values between configuration file and command lien arguments

# Changes
- Remove type `NamePathBinding` as it was not necessary anymore
- Remove custom implementation for flag parsing and replace it with [pflag](https://pkg.go.dev/github.com/spf13/pflag) to have a POSIX compliant CLI behaviour.
- Refactor actions and other functions to take `Configuration` as an input instead of `Arguments`. This was made to provide a validation, parsing and flag merging ability to exeiac.
- (minor) Add godoc documentation for functions and types
- (minor) Move `BadArg` error to the `error.go` file to keep things consistent. We might end up creating a `error` package down the line.
- (minor) Move functions around; `EnrichBricks()` & `ValidateConfiguration()` (used to be `validateBrickNames()`) to be specific
- (minor) Use a different style for long function signatures
```go
// We went from
func (inter *MyInterface) MyLongFunctionSignature(
  first FirstArgumentType,
  first FirstArgumentType,
  first FirstArgumentType) (value ReturnValueType, err error) {

  //My function's body
 //...
}

// To
func (inter *MyInterface) MyLongFunctionSignature(
  first FirstArgumentType,
  first FirstArgumentType,
  first FirstArgumentType,
) (
value ReturnValueType,
err error,
) {
  //My function's body
 //...
}
```

# Business changes
- Change the `brick.yml` format to have an array of string for the `bricks_specifiers` and `other_options` keys to better reflect its usage and internal typing
- Change separator for brick specifiers (`--brick-specifiers`). Used POSIX standard one (comma `,`) instead of `+`. If keeping `+` and deviating from CLI standard is truly wanted, it'll be done in a separate commit.

### CODE REVIEW

File: src/main.go [hunk 1]
```diff
@@ -2,34 +2,72 @@ package main
...
+// c.f. https://go.dev/doc/effective_go#init
+func init() {
+	flag.StringSliceVarP(&arguments.BricksSpecifiers, "bricks-specifiers", "s", []string{"selected"},
+		fmt.Sprintf("A list of comma separated specifiers. Includes: %v", exargs.AvailableBricksSpecifiers))
+
+	flag.StringVarP(&arguments.ConfigurationFile, "configuration-file", "c", "/etc/exeiac/conf.yml",
+		fmt.Sprint("A path the a valid configuration file"))
+
+	flag.BoolVarP(&arguments.Interactive, "interactive", "i", false,
+		fmt.Sprint("Wheither or not to execute the module interactively"))
```
Comment (half-shell) [inline src/main.go:25]: Change to being non-interactive

File: src/main.go [hunk 2]
```diff
@@ -2,34 +2,72 @@ package main
...
+
+	defaultRooms := make(map[string]string)
+	flag.StringToStringVarP(&arguments.Rooms, "rooms", "r", defaultRooms,
+		fmt.Sprint("A set of key/value pairs with the key being the name of the room, and the value its absolute path. This is useful if you want to setup a room on the command line. It uses the configuration file's rooms declaration otherwise."))
+
+	// TODO(half-shell): Defaulting to "." does not seem right.
+	// What are the options for this flag? It does not seem to be used anywhere for now.
+	// TODO(half-shell): The shorthand option does not really match a pattern. How about "-os" intead?
+	flag.StringVarP(&arguments.OutputSpecifier, "output-specifier", "S", ".",
+		fmt.Sprint("A kinf of output specifier"))
```
Comment (half-shell) [inline src/main.go:42]: Remove

File: src/arguments/configuration.go [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+		sb.WriteString(t.Field(i).Name)
+		sb.WriteString(": ")
+		sb.WriteString(fmt.Sprintf("%v", v.Field(i).Interface()))
+		sb.WriteString("\n")
+	}
+
+	return sb.String()
+}
+
+func CreateConfiguration(confFilePath string) (configuration Configuration, err error) {
```
Comment (half-shell) [inline src/arguments/configuration.go:60]: Make private

File: src/main.go [hunk 4]
```diff
@@ -39,24 +77,24 @@ func main() {
 
 	// get bricks selected
 	var bricks exinfra.Bricks
-	bricks, err = infra.GetBricksFromNames(args.BricksNames)
+	bricks, err = infra.GetBricksFromNames(arguments.BricksNames)
```
Comment (half-shell) [inline src/main.go:80]: Update to configuration

**Review by arthur91f — APPROVED** [review]

**Pull request merged**